### PR TITLE
Add CI visibility to system-probe kitchen tests

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Adds Datadog CI visibility to the system-probe kitchen tests run on many different Linux kernels via uploading junit XML files. 

### Motivation

Improved tracking of specific test failure rate and duration.

### Additional Notes

This PR makes several changes to how system-probe tests are run:

1. `gotestsum` is used run the tests, generating a junit XML file for upload to Datadog CI. The results are tagged with the platform (distro + distro version, example `ubuntu-22.04`), architecture, kernel version, and bundle (how the eBPF code is built: `prebuilt`, `runtime`, or `co-re`).
2. A JSON file is also produced containing the concatenated output of `test2json` (used when running `go test -json`).
3. The output format is significantly compacted, only generating output for tests that are skipped or fail. (See example below). The full accounting of which tests ran is available via the xml files, json files, or the Datadog CI app.
4. The verifier from test-kitchen's perspective *always passes*. This is due to a current bug in test-kitchen, which only downloads files on success. I need to download the XML and JSON files no matter success or failure. Thus, I wrote a small python task to examine the JSON output, and correctly fail the Gitlab job if any tests fail. This also has the benefit of outputting a compact listing of all failing tests for each platform and bundle combination in the Gitlab job output.
5. I changed how the eBPF mode of building is chosen for tests. Rather than relying upon test-specific environment variables, I now use standard configuration environment variables to control the behavior.

Screen Shot 2022-11-09 at 10 18 14 AM
Screen Shot 2022-11-09 at 10 19 35 AM
Screen Shot 2022-11-09 at 10 18 35 AM

### Possible Drawbacks / Trade-offs

The automatic uploading test results only functions when the system-probe kitchen jobs are automatically triggered, due to specific file/directory changes. If you manually run the jobs, you must also manually run the upload job afterwards. I do not know how the CI app would treat duplicate upload results from the same job/pipeline.

This does not currently affect:
1. Windows tests.
2. CWS tests (but they can be added afterwards with little change)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by clamoriniere — APPROVED** [review]
/LGTM for #container-integrations files

**Comment by guyarb** [comment]
commenting on the picture in the description
There are many tests that are nested, and the failure happens only in the inner test
we might want to ignore the two tests above to reduce noises

File: .gitlab/functional_test_junit_upload.yml
Comment (hmahmood) [inline .gitlab/functional_test_junit_upload.yml:20]: Is this for debugging only? Still needed?

**Comment by brycekahle** [comment]
> we might want to ignore the two tests above to reduce noises

This is tricky, because technically you can fail outside of the nested tests. Go reports each level of nesting separately, so that is what the junit XML has.

**Review by guyarb — APPROVED** [review]
LGTM

File: tasks/system_probe.py
Comment (hmahmood) [inline tasks/system_probe.py:611]: Is `cmd/test2json` not included in the PR?

**Review by hmahmood — COMMENTED** [review]
Where are we getting `gotestsum` from?

File: .gitlab/functional_test_junit_upload.yml
Comment (brycekahle) [reply .gitlab/functional_test_junit_upload.yml:20]: Confusingly, this turns OFF logging. It is meant to hide the output from the ssm command. This is copied verbatim from `source_test_junit_upload.yml`.

File: tasks/system_probe.py
Comment (brycekahle) [reply tasks/system_probe.py:611]: It is part of the Go distribution, so always available.

**Comment by brycekahle** [comment]
> Where are we getting gotestsum from?

It is part of the tools that are installed with `inv install-tools`. The main agent tests were already using it.

File: tasks/system_probe.py
Comment (hmahmood) [reply tasks/system_probe.py:611]: I thought it was doing the opposite :-)

File: .gitlab/functional_test_junit_upload.yml
Comment (hmahmood) [reply .gitlab/functional_test_junit_upload.yml:20]: I thought it was doing the opposite :-)

**Review by hmahmood — APPROVED** [review]

File: tasks/system_probe.py
Comment (ISauve) [inline tasks/system_probe.py:530]: nice, this is  much cleaner way of setting these configuration values 👍

**Review by ISauve — APPROVED** [review]
🎉

**Review by hush-hush — APPROVED** [review]

**Review by bkabrda — COMMENTED** [review]
:wave: this looks really good. I've just left one minor question inline - once we get that sorted out, I can approve.

**Review by paulcacheux — APPROVED** [review]

File: test/kitchen/platforms.json
Comment (brycekahle) [reply test/kitchen/platforms.json:85]: This is for running kitchen tests locally. This is especially helpful when developing, as I don't have to wait for the CI.

Comment (bkabrda) [reply test/kitchen/platforms.json:85]: Ok, I guess this is fine in the Vagrant section. Thanks!

**Review by bkabrda — APPROVED** [review]

**Pull request merged**
